gamla/functional_generic.py

```diff
@@ -154,6 +155,14 @@ def ternary_inner(*args, **kwargs):
 curried_ternary = ternary
 
 
+def when(condition, f_true):
```

Comment:
isn't this like `do_if`?
better name nonetheless

Reply:
no. do_if does not return the side_effect, it returns the input (that is why it's called `do`)